packages/webapp/src/containers/Animals/Inventory/useAnimalOrBatchRemoval.ts

```diff
@@ -174,32 +174,26 @@ const useAnimalOrBatchRemoval = (
     }
   };
 
-  const getFinalizedTaskIdsSet = useCallback(() => {
-    return new Set([...completedTasks, ...abandonedTasks].map(({ task_id }) => task_id));
+  const getFinalizedTasks = useCallback(() => {
+    return Array.from(new Set([...completedTasks, ...abandonedTasks]));
   }, [completedTasks, abandonedTasks]);
 
   const hasFinalizedTasks = useMemo(() => {
     if (!removalModalOpen) {
       return false;
     }
 
-    const selectedInventoryIdsSet = new Set(selectedInventoryIds);
-    const finalizedTaskIdsSet = getFinalizedTaskIdsSet();
-
-    let inventoryFoundCount = 0;
-    for (let { id, tasks } of animalTasksWithInventoryIds) {
-      if (selectedInventoryIdsSet.has(id)) {
-        inventoryFoundCount++;
-
-        if (tasks.some(({ task_id }) => finalizedTaskIdsSet.has(task_id))) {
-          return true;
-        }
-      }
-      // Stop iterating once all selected inventory IDs have been processed.
-      if (inventoryFoundCount === selectedInventoryIdsSet.size) {
-        return false;
-      }
-    }
+    const finalizedTasks = getFinalizedTasks();
+
+    return selectedInventoryIds.some((id) =>
```

Comment:
@antsgar I think this would do it -- but its not pretty:

return selectedInventoryIds.some((animalOrBatchId) => {
        const {id, kind} = parseInventoryId(animalOrBatchId);
        return finalizedTasks.filter(
          ({ animals, animal_batches }: { animals: Animal[]; animal_batches: AnimalBatch[] }) => {
            const animalIds = animals.map(({ id }) => `${id}`);
            const batchIds = animal_batches.map(({ id }) => `${id}`);
            return kind === AnimalOrBatchKeys.ANIMAL ? animalIds.includes(`${id}`) : batchIds.includes(`${id}`);
          },
        ).length > 0;
      }
    );

Reply:
Made a PR for it if helpful -- .. removed redundant `> 0`